lib/web_ui/lib/src/engine/canvaskit/surface.dart

```diff
@@ -205,70 +206,38 @@ class Surface {
         final ui.Size newSize = size * 1.4;
         _surface?.dispose();
         _surface = null;
-        htmlCanvas!.width = newSize.width;
-        htmlCanvas!.height = newSize.height;
+        if (Surface.offscreenCanvasSupported) {
+          _offscreenCanvas!.width = newSize.width;
+          _offscreenCanvas!.height = newSize.height;
+        } else {
+          _canvasElement!.width = newSize.width;
+          _canvasElement!.height = newSize.height;
+        }
         _currentCanvasPhysicalSize = newSize;
         _pixelWidth = newSize.width.ceil();
         _pixelHeight = newSize.height.ceil();
-        _updateLogicalHtmlCanvasSize();
       }
     }
 
     // Either a new context is being forced or we've never had one.
     if (_forceNewContext || _currentCanvasPhysicalSize == null) {
       _surface?.dispose();
       _surface = null;
-      _addedToScene = false;
       _grContext?.releaseResourcesAndAbandonContext();
       _grContext?.delete();
       _grContext = null;
 
       _createNewCanvas(size);
       _currentCanvasPhysicalSize = size;
-    } else if (window.devicePixelRatio != _currentDevicePixelRatio) {
-      _updateLogicalHtmlCanvasSize();
-    }
+    } else if (window.devicePixelRatio != _currentDevicePixelRatio) {}
```

Comment:
I guess this entire else block should be removed? (It's just an empty block now)

Reply:
Done.